deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,297 @@
+    ...               n_distance=100,
+    ...               learning_rate=1.0,
+    ...               mode="regression")
+    >>> loss = model.fit(data, nb_epoch=250)
+    >>> pred = model.predict(data)
+
+    References
+    ----------
+    .. [1] Schütt, Kristof T., et al. "Quantum-chemical insights from deep
+        tensor neural networks." Nature communications 8.1 (2017): 1-8.
+
+    """
+
+    def __init__(self,
+                 n_tasks: int,
+                 n_embedding: int = 30,
+                 n_hidden: int = 100,
+                 n_distance: int = 100,
+                 distance_min: float = -1,
+                 distance_max: float = 18,
+                 output_activation: bool = True,
+                 mode: str = "regression",
+                 dropout: float = 0.0,
+                 n_steps: int = 2,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        n_embedding: int (default 30)
+            Number of features per atom.
+        n_hidden: int (default 100)
+            Number of features for each molecule after DTNNStep
+        n_distance: int (default 100)
+            granularity of distance matrix
+            step size will be (distance_max-distance_min)/n_distance
+        distance_min: float (default -1)
+            minimum distance of atom pairs (in Angstrom)
+        distance_max: float (default = 18)
+            maximum distance of atom pairs (in Angstrom)
+        output_activation: bool (default True)
+            determines whether an activation function should be apply  to its output.
+        mode: str (default "regression")
+            Only "regression" is currently supported.
+        dropout: float (default 0.0)
+            the dropout probablity to use.
+        n_steps: int (default 2)
+            Number of DTNNStep Layers to use.
+
+        """
+        if dropout < 0 or dropout > 1:
+            raise ValueError("dropout probability has to be between 0 and 1, "
+                             "but got {}".format(dropout))
+        model = DTNN(n_tasks=n_tasks,
+                     n_embedding=n_embedding,
+                     n_hidden=n_hidden,
+                     n_distance=n_distance,
+                     distance_min=distance_min,
+                     distance_max=distance_max,
+                     output_activation=output_activation,
+                     mode=mode,
+                     dropout=dropout,
+                     n_steps=n_steps)
+        if mode not in ['regression']:
+            raise ValueError("Only 'regression' mode is currently supported")
+        super(DTNNModel, self).__init__(model, L2Loss(), ["prediction"],
+                                        **kwargs)
+
+    def compute_features_on_batch(self, X_b):
+        """Computes the values for different Feature Layers on given batch
+
+        A tf.py_func wrapper is written around this when creating the
```

Comment:
This comment references TF which we don't use here (pytorch)

Reply:
Removed It